src/main/java/entity/ForumEntry.java

```diff
@@ -0,0 +1,80 @@
+package entity;
+
+import java.time.LocalDateTime; 
+
+public abstract class ForumEntry {
+    private String entryID;
+    private User author;
+    private Content content;
+    private LocalDateTime postedDate;
+    private LocalDateTime editDate; 
+    private int likes;
+    private int dislikes;
+
+    public ForumEntry(String entryID, User author, Content content, LocalDateTime postedDate,
+                      LocalDateTime editDate, int likes, int dislikes) {
+        this.entryID = entryID;
+        this.author = author;
+        this.content = content;
```

Comment:
I think it might be a good idea to have some validation here e.g. if content = null , throw an error?

Reply:
I feel like that's the job in the controller, if everything is successful/valid, we create a post and prepare a success view. Else we don't and prepare a failure view. I believe entities shouldn't handle logics.

Reply:
Oh right right yes